libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -346,6 +342,56 @@ string ABIFunctions::cleanupFunction(Type const& _type, bool _revertOnFailure)
 	});
 }
 
+string ABIFunctions::validatorFunction(Type const& _type, bool _revertOnFailure)
+{
+	string functionName = string("validator_") + (_revertOnFailure ? "revert_" : "assert_") + _type.identifier();
+	return createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+			function <functionName>(value) {
+				if iszero(<condition>) { <failure> }
+			}
+		)");
+		templ("functionName", functionName);
+		if (_revertOnFailure)
+			templ("failure", "revert(0, 0)");
+		else
+			templ("failure", "invalid()");
+
+		switch (_type.category())
+		{
+		case Type::Category::Address:
+		case Type::Category::Integer:
+		case Type::Category::RationalNumber:
+		case Type::Category::Bool:
+		case Type::Category::FixedPoint:
+		case Type::Category::Function:
+		case Type::Category::Array:
+		case Type::Category::Struct:
+		case Type::Category::Mapping:
```

Comment:
Added "mapping".

Reply:
Shouldn't we rather *remove* ``Struct`` and ``Array``? Not that it matter's too much - ``cleanupFunction`` will assert anyways, but removing them here would yield the better assertion message... but fine either way.

Reply:
No, it will only assert if they are not stored in storage!